### PR TITLE
Delete dmd/src/dmd

### PR DESCRIPTION
**Pull request by RazvanN7**
After modifying all the scrips to use dmd/generated/os/build/model/dmd instead of dmd/src/dmd, we can delete the dmd/src/dmd binary.

### CODE REVIEW

File: src/posix.mak [hunk 1]
```diff
@@ -351,9 +351,6 @@ $G/backend.a: $(G_OBJS)
 $G/dmd_frontend: $(FRONT_SRCS) $D/gluelayer.d $(ROOT_SRCS) $G/newdelete.o $(STRING_IMPORT_FILES) $(HOST_DMD_PATH)
 	CC=$(HOST_CXX) $(HOST_DMD_RUN) -of$@ $(MODEL_FLAG) -vtls -J$G -J../res -L-lstdc++ $(DFLAGS) $(filter-out $(STRING_IMPORT_FILES) $(HOST_DMD_PATH),$^) -version=NoBackend
 
-dmd: $G/dmd $G/dmd.conf
-	cp $< .
-
```
Comment (wilzbach) [inline src/posix.mak]: Could you please remove the src/dmd.conf target as well? Thanks!

**Comment by RazvanN7** [comment]
CC @braddr , @CyberShadow , @MartinNowak

**Comment by CyberShadow** [comment]
In win32.mak:

```
G = $(GEN)\$(OS)$(MODEL)
```
`OS` is not defined anywhere.

**Comment by CyberShadow** [comment]
Also, the directory structure is very different on Windows and POSIX.

**Comment by CyberShadow** [comment]
Noting that to reliably use the new paths, one needs to either glob, or either invoke or reimplement osmodel.mak in their code. This is a regression in complexity. Perhaps it would be an improvement to allow specifying the desired output path of the DMD binary as a Makefile parameter.

**Comment by RazvanN7** [comment]
@CyberShadow From what I could deduce, the directory hierarchies are very different when it comes to posix vs windows builds. For example, if you look at phobos : the linux version has a generated/os/build/model hierarchiy, while the windows version employs a totally different hierarchy (there is a src folder where all the sources are kept - dmd, druntime, phobos - alltogheter and a bin directory where libraries and binaries are generated). Seeing this inconsistency, I thought that the build chains are considered to be different and as a result the hierarchies are different. I will add in win32.mak the os variable which will be set to windows and the build variable which will be set to release, but I don't think that will make any difference.

**Comment by CyberShadow** [comment]
Yep, all true. But if we are to define new hierarchies, might as well make them consistent with themselves, right?

File: src/posix.mak (hunk 1 above)
Comment (RazvanN7) [inline src/posix.mak]: I assume you were referring to $G/dmd.conf

**Comment by CyberShadow** [comment]
> Addressed issues in comments

How come @MartinNowak is the committer (or author?) of that commit? And was that meant to be an amended commit?

**Comment by RazvanN7** [comment]
@CyberShadow It was intended to be an amended commit, but I have no idea why Martin appeared to coauthor the commit. I rebased and tried to commit again but now it looks like dlang-bot is the coauthor of this commit

**Comment by CyberShadow** [comment]
I think you might have rebased/amended too far and included the previous merge commit on master.

**Comment by CyberShadow** [comment]
I'd suggest trying to amend with `--reset-author`.

**Comment by RazvanN7** [comment]
@CyberShadow Looking through my commit logs I saw that the first commit on this branch accidentally included --amend, but --reset-author solved the case. Thanks

**Comment by CyberShadow** [comment]
OK, though you still should rebase on master. Right now your last commit has two parents, and is a merge commit between `master` and the tip of PR #6572.

**Comment by RazvanN7** [comment]
@andralex @CyberShadow @wilzbach I think this is ready to be merged. Next stop: dmd -> ddmd

**Comment by wilzbach** [comment]
@RazvanN7 I think sth went wrong during your rebase - have a look at the changes tab

**Comment by RazvanN7** [comment]
@wilzbach It seems that circleci needs a dmd.conf in generated

**Comment by andralex** [comment]
Well things are failing for the time being. How do we kickstart this?

**Comment by RazvanN7** [comment]
@MartinNowak After this PR : https://github.com/dlang/druntime/pull/1779 one would expect that the autotester uses dmd in generated/os/release/model. Why isn't that happening?

**Comment by RazvanN7** [comment]
@wilzbach I had to keep the dmd.conf target otherwise circleci and travis would fail.

**Comment by andralex** [comment]
@RazvanN7 please take a look at the auto-tester (https://auto-tester.puremagic.com/pull-history.ghtml?projectid=1&repoid=1&pullid=6595), seems to still need ../src/dmd. E.g. from Darwin_64_64:

```
../dmd/src/dmd -conf= -c -o- -Isrc -Iimport -Hfimport/core/sync/barrier.di src/core/sync/barrier.d
make: ../dmd/src/dmd: No such file or directory
```

**Comment by andralex** [comment]
@MartinNowak @braddr @CyberShadow @wilzbach summoning the wizards for a puzzling matter with the autotester.

I see the autotester fails on e.g. FreeBSD_32 with:

```
../dmd/src/dmd -conf= -c -o- -Isrc -Iimport -Hfimport/core/sync/barrier.di src/core/sync/barrier.d
gmake: ../dmd/src/dmd: Command not found
posix.mak:146: recipe for target 'import/core/sync/barrier.di' failed
```

This looks trivial, enough, except https://github.com/dlang/druntime/blob/master/posix.mak has already been changed to use generated/ instead of src/ for locating the dmd binary, and in fact even the line given in the error message (146) is off by one (should be 147). That suggests the druntime makefile run by the autotester is older than the one in master.

What could cause this?

**Comment by rainers** [comment]
> What could cause this?

The auto-tester passes it explicitly on the command line: https://github.com/braddr/at-client/blob/master/src/do_build_druntime.sh#L15. Same for phobos.

**Comment by andralex** [comment]
@rainers thanks! https://github.com/braddr/at-client/pull/1

**Comment by andralex** [comment]
@wilzbach is this related to your recent work?

**Comment by andralex** [comment]
@RazvanN7 please rebase so we keep this in good shape until we sort it out

**Comment by RazvanN7** [comment]
I think that we can close this since it is out of sync and @wilzbach 's PR does the same thing and much more.

**Pull request closed without merging**